spec/rubocop/cop/lint/self_assignment_spec.rb

```diff
@@ -129,4 +129,56 @@
       foo &&= bar
     RUBY
   end
+
+  it 'registers an offense when using attribute self-assignment' do
+    expect_offense(<<~RUBY)
+      foo.bar = foo.bar
+      ^^^^^^^^^^^^^^^^^ Self-assignment detected.
+    RUBY
+  end
+
+  it 'does not register an offense when using attribute assignment with different attributes' do
+    expect_no_offenses(<<~RUBY)
+      foo.bar = foo.baz
+    RUBY
+  end
+
+  it 'does not register an offense when using attribute assignment with different receivers' do
+    expect_no_offenses(<<~RUBY)
+      bar.foo = baz.foo
+      foo.bar = foo.bar + 1
+    RUBY
+  end
+
+  it 'does not register an offense when using attribute assignment with extra expression' do
+    expect_no_offenses(<<~RUBY)
+      foo.bar = foo.bar + 1
+    RUBY
+  end
+
+  it 'registers an offense when using []= self-assignment' do
+    expect_offense(<<~RUBY)
+      foo["bar"] = foo["bar"]
+      ^^^^^^^^^^^^^^^^^^^^^^^ Self-assignment detected.
+    RUBY
+  end
+
+  it 'does not register an offense when using []= assignment with different keys' do
+    expect_no_offenses(<<~RUBY)
+      foo["bar"] = foo["baz"]
+    RUBY
+  end
+
+  it 'does not register an offense when using []= assignment with different receivers' do
+    expect_no_offenses(<<~RUBY)
+      bar["foo"] = baz["foo"]
+      foo["bar"] = foo["bar"] + 1
```

Comment:
`foo["bar"] = foo["bar"] + 1` is the same as L181. Is it left out intentionally?